typescript/lib/services/service_context.ts

```diff
@@ -0,0 +1,129 @@
+    public execCommand(command: string[]): [number, [], Error] { //TODO - [int32, pointer to []byte (comment), error ] ; is is not []`type` like []string
+        let serviceId = this.serviceId;
+        let args = constructors.newGetExecCommandArgs(serviceId, command);
+        let request: grpc.requestCallback<bindingsJs.ExecCommandResponse>; //TODO - Tried to implement callback through examples on Google while also respecting the actual Callback function inside grpc
+
+        let resp, err = this.client.execCommand(args, request); //TODO - does error checking work here ; check the return type of execCommand, it's not an ExecCommandResponse ? 
+        
+        if (err != null) {
+            return [0, null, new Error( //TODO - Error type (do I need callback?)
+			    "An error occurred executing command '%v' on service '%v'", //TODO - I might not be able keep the '%v', but how I throw the actual error
+            )]; //TODO (comment) cannot return multipe values in typescript function, need to wrap in array or object
+        }
+        return [resp.ExitCode, resp.LogOutput, null]; //TODO need to make sure resp is ExecCommandResponse if I want to return this ; (comment) resp.LogOuput was a pointer ; returning multiple values as an array
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public generateFiles(filesToGenerateSet: Set<string>): [Map<string, GeneratedFileFilepaths>, Error] {
+        let serviceId = this.serviceId;
+        let fileGenerationOpts = new Map(); //TODO (comment) Key: string & Value: core_api_bindings.FileGenerationOptions{}
+        for (let fileId in filesToGenerateSet) {
+            fileGenerationOpts[fileId] = new bindingsJs.FileGenerationOptions(); //TODO (comment) this was a pointer
+            fileGenerationOpts[fileId].setFileTypeToGenerate(bindingsJs.FileGenerationOptions.FileTypeToGenerate.FILE);
+        }
+
+        let args = constructors.newGetGenerateFilesArgs(serviceId, fileGenerationOpts);
+        let request: grpc.requestCallback<bindingsJs.GenerateFilesResponse>;
+
+        let resp, err = this.client.generateFiles(args, request); //TODO - returns grpc.ClientUnaryCall, doesn't directly return error & not possible to return multiple values directly from function
+        if (err != null){
+            return [null, new Error("An error occurred generating files using args: %+v")]; //TODO - no %+v maybe, but should be throwing actual error
+        }
+        let generatedFileRelativeFilepaths = resp.GeneratedFileRelativeFilepaths;
+
+        let result = new Map(); // TODO Key:string & Value:GeneratedFileFilepaths{} - type not needed in typescript
+        for (let fileId in filesToGenerateSet) {
+            //TODO - need to add check if fileID is inside map so that it it either manipulates the map or sens an error
+            var relativeFilepath;
+            if (generatedFileRelativeFilepaths.has(fileId)) {
```

Comment:
This doesn't follow "eject early, eject often": https://github.com/kurtosis-tech/kurtosis-internal-docs/blob/master/general-style-guide.md#default-to-eject-early-eject-often

A better way to write this:
```
if (!generatedFileRelativeFilepaths.has(fileId)) {
                return [null, new Error(
                    "No filepath (relative to test volume root) was returned for file '%v', even though we requested it; this is a Kurtosis bug", //TODO - no %v maybe
                    )];
}
relativeFilepath = generatedFileRelativeFilepaths[fileId];
```

(which is I'm pretty sure how the Go code does it)

Reply:
Also, quick aside - I strongly prefer this grouping of braces, which makes indentation very clear:
```
if {
    .....
} else if {
    .....
} else {
    ....
}
```
(Trivia: this is called "cuddling" the braces)

Reply:
Ah we want to return the error first to eject early!